apis/kubekey/v1alpha2/etcd_types.go

```diff
@@ -0,0 +1,48 @@
+/*
+ Copyright 2021 The KubeSphere Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package v1alpha2
+
+const (
+	KubeKey  = "kubekey"
+	Kubeadm  = "kubeadm"
+	External = "external"
+)
+
+type EtcdCluster struct {
+	// Type of etcd cluster, can be set to 'kubekey' 'kubeadm' 'external'
+	Type string `yaml:"type" json:"type,omitempty"`
+	// ExternalEtcd describes how to connect to an external etcd cluster when type is set to external
+	External         ExternalEtcd `yaml:"external" json:"external,omitempty"`
```

Comment:
What do you think about removing the `--certificates-dir` flag? I think it's more consistent to use the `certFile` in the config to specify all cases.
BTW, may need to change the name `external`.

Reply:
When I deal with the logic here, I'm also confused about why there is this parameter `--certificates-dir`. It should have been considered at that time that there might be a need to specify the certificate generation path, but now it seems that this parameter can be removed.

This structure is defined by referring to kubeadm. 
https://pkg.go.dev/k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1beta2#Etcd

Reply:
Done